extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -187,59 +187,48 @@ describes.realWin('amp-story', {
-    });
-  });
-
-  // TODO(#11639): Re-enable this test.
-  it.skip('should update progress bar when switching pages', () => {
-    const impl = element.implementation_;
-    const count = 10;
-    const index = 2;
-
-    const page = win.document.createElement('div');
-
-    const updateProgressBarStub =
-        sandbox.stub(impl.systemLayer_, 'updateProgressBar').callsFake(NOOP);
-
-    appendEmptyPage(element, /* opt_active */ true);
-
-    sandbox.stub(impl, 'getPageCount').returns(count);
-    sandbox.stub(impl, 'getPageIndex').withArgs(page).returns(index);
-
-    impl.switchTo_(page);
+  it('should return a valid page index', () => {
+    createPages(story.element, 4, ['cover', 'page-1', 'page-2', 'page-3']);
+    return story.layoutCallback()
+        .then(() => {
+        // Getting all the AmpStoryPage objets.
+          const pageElements =
+            story.element.getElementsByTagName('amp-story-page');
+          const pages = Array.from(pageElements).map(el => el.getImpl());
 
-    // first page is not counted as part of the progress
-    expect(updateProgressBarStub).to.have.been.calledWith(index, count - 1);
+          return Promise.all(pages);
+        })
+        .then(pages => {
+        // Only the first page should be active.
+          for (let i = 0; i < pages.length; i++) {
+            expect(story.getPageIndex(pages[i])).to.equal(i);
+          }
+        });
   });
 
-  // TODO(#11639): Re-enable this test.
-  it.skip('should pause/resume pages when switching pages', () => {
-    const impl = element.implementation_;
-    const pages = createPages(element, 5);
-    impl.schedulePause = sandbox.spy();
-    impl.scheduleResume = sandbox.spy();
-
-    const oldPage = pages[0];
-    const newPage = pages[1];
-
-    element.build();
 
-    return impl.switchTo_(newPage).then(() => {
-      expect(impl.schedulePause).to.have.been.calledWith(oldPage);
-      expect(impl.scheduleResume).to.have.been.calledWith(newPage);
-    });
+  it('should pause/resume pages when switching pages', () => {
+    createPages(story.element, 2, ['cover', 'page-1']);
+    return story.layoutCallback()
+        .then(() => {
+        // Getting all the AmpStoryPage objets.
+          const pageElements =
+            story.element.getElementsByTagName('amp-story-page');
+          const pages = Array.from(pageElements).map(el => el.getImpl());
+          return Promise.all(pages);
+          const oldPage = pageElements[0].implementation_;
+          const newPage = pageElements[1].implementation_;
+          const pauseOldPageStub = sandbox.stub(oldPage, 'pauseCallback');
+          const resumeNewPageStub = sandbox.stub(newPage, 'resumeCallback');
+          story.switchTo_('page-1').then(() => {
+            expect(pauseOldPageStub).to.have.been.calledOnce;
+            expect(resumeNewPageStub).to.have.been.calledOnce;
+          });
+        });
   });
 
   // TODO(#11639): Re-enable this test.
```

Comment:
could you also remove this comment please?

Reply:
I actually haven't figured out that one yet, so I was supposed to skip it as well instead of accidentally enabling it again. Sorry for the confusion! I readded the skip so the comment makes sense